### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
**What:**

This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

**Why:**

Typos hide bugs, harm usability, making it harder to search for things

**How:**

The misspellings have been reported at https://github.com/jsoref/facebookincubator-dns/actions/runs/5036537407#summary-13637046995

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/facebookincubator-dns/actions/runs/5036537561#summary-13637047289

**Risks:**

**Checklist**:

- [ ] Added tests, if you've added code that should be tested
- [ ] Updated the documentation, if you've changed APIs
- [x] Ensured the test suite passes
- [x] Made sure your code lints
- [x] Completed the Contributor License Agreement ("CLA")

### CODE REVIEW

File: dnsrocks/cgo-rocksdb/backup.go
Comment (jsoref) [inline dnsrocks/cgo-rocksdb/backup.go:1]: it's possible this is a third party module. I can send PRs to other places instead...

File: dnsrocks/dnsdata/rdb/rdb_iteratorPool.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ import (
 // NumberOfIterators is empirically found number of iterators which make sense to keep in pool
 const NumberOfIterators int = 15
 
-// IteratorPool allows RDB interators reuse. Iterator creation is happen to be pretty costly operation
+// IteratorPool allows RDB iterators reuse. Iterator creation is happen to be pretty costly operation
```
Comment (jsoref) [inline dnsrocks/dnsdata/rdb/rdb_iteratorPool.go:25]: probably `iterator`...

File: dnsrocks/logger/dnstap_logger.go [hunk 2]
```diff
@@ -49,18 +49,18 @@ var (
 	localRand = db.NewRand()
 )
 
-// DNSTapLoggger logs to dnstap output
-type DNSTapLoggger struct {
+// DNSTapLogger logs to dnstap output
+type DNSTapLogger struct {
```
Comment (jsoref) [inline dnsrocks/logger/dnstap_logger.go:53]: It's possible this is a public interface and thus can't be changed...

**Comment by leoleovich** [comment]
Please squash all the commits into 1.

**Pull request closed without merging**